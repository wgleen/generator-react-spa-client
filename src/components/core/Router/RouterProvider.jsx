import React from 'react'
import PropTypes from 'prop-types'
import { BrowserRouter as Router } from 'react-router-dom'

const RouterProvider = (props) => {
  const { children } = props

  return <Router>{children}</Router>
}

RouterProvider.defaultProps = {
  children: null
}

RouterProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.node,
    PropTypes.string
  ])
}

export default RouterProvider
