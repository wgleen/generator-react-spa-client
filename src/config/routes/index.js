const routes = {
  root: {
    path: '/'
  }
}

export default routes
