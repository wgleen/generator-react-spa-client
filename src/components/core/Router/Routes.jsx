import React, { lazy, Suspense } from 'react'
import { Route, Switch } from 'react-router-dom'
import routes from '../../../config/routes'

const Home = lazy(() => import('../../pages/Home/Home'))

const Routes = () => (
  <Suspense fallback="Loading">
    <Switch>
      <Route exact path={routes.root.path} component={Home} />
    </Switch>
  </Suspense>
)

export default Routes
