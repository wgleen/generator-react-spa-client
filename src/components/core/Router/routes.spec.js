import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import RouterProvider from './RouterProvider'
import routes from '../../../config/routes'
import Routes from './Routes'

describe('<Routes />', () => {
  it('should render correctly', () => {
    const component = mount(
      <MemoryRouter>
        <Routes />
      </MemoryRouter>
    )

    expect(component).toMatchSnapshot()

    component.unmount()
  })

  describe('when the current page is equal to routes.root.path', () => {
    it.skip('should render the Home component', () => {
      const component = mount(
        <MemoryRouter initialEntries={[routes.root.path]} initialIndex={0}>
          <Routes />
        </MemoryRouter>
      )

      expect(component.find('Home').exists()).toBeTrue()

      component.unmount()
    })
  })
})
