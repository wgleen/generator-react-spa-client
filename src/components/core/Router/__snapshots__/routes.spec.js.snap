// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> should render correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "2ee6kx",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "2ee6kx",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Routes>
      <Suspense
        fallback="Loading"
      >
        <Switch>
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_init": [Function],
                "_payload": Object {
                  "_result": Promise {},
                  "_status": 0,
                },
              }
            }
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "key": "2ee6kx",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          />
        </Switch>
        Loading
      </Suspense>
    </Routes>
  </Router>
</MemoryRouter>
`;
